ipaserver/plugins/cert.py

```diff
@@ -1580,6 +1585,8 @@ def _ldap_search(self, all, pkey_only, no_members, **options):
 
     def execute(self, criteria=None, all=False, raw=False, pkey_only=False,
                 no_members=True, timelimit=None, sizelimit=None, **options):
+        self.ca_enabled = self.api.Command.ca_is_enabled()['result']
```

Comment:
`ca_enabled` seems to be locked in Python 2:
```
[Mon Oct 30 16:23:02.474784 2017] [wsgi:error] [pid 4859] [remote 172.17.0.2:47078] ipa: ERROR: non-public: AttributeError: locked: cannot set cert_find.ca_enabled to True
[Mon Oct 30 16:23:02.474810 2017] [wsgi:error] [pid 4859] [remote 172.17.0.2:47078] Traceback (most recent call last):
[Mon Oct 30 16:23:02.474813 2017] [wsgi:error] [pid 4859] [remote 172.17.0.2:47078]   File "/usr/lib/python2.7/site-packages/ipaserver/rpcserver.py", line 370, in wsgi_execute
[Mon Oct 30 16:23:02.474815 2017] [wsgi:error] [pid 4859] [remote 172.17.0.2:47078]     result = command(*args, **options)
[Mon Oct 30 16:23:02.474817 2017] [wsgi:error] [pid 4859] [remote 172.17.0.2:47078]   File "/usr/lib/python2.7/site-packages/ipalib/frontend.py", line 450, in __call__
[Mon Oct 30 16:23:02.474820 2017] [wsgi:error] [pid 4859] [remote 172.17.0.2:47078]     return self.__do_call(*args, **options)
[Mon Oct 30 16:23:02.474822 2017] [wsgi:error] [pid 4859] [remote 172.17.0.2:47078]   File "/usr/lib/python2.7/site-packages/ipalib/frontend.py", line 478, in __do_call
[Mon Oct 30 16:23:02.474824 2017] [wsgi:error] [pid 4859] [remote 172.17.0.2:47078]     ret = self.run(*args, **options)
[Mon Oct 30 16:23:02.474826 2017] [wsgi:error] [pid 4859] [remote 172.17.0.2:47078]   File "/usr/lib/python2.7/site-packages/ipalib/frontend.py", line 800, in run
[Mon Oct 30 16:23:02.474829 2017] [wsgi:error] [pid 4859] [remote 172.17.0.2:47078]     return self.execute(*args, **options)
[Mon Oct 30 16:23:02.474831 2017] [wsgi:error] [pid 4859] [remote 172.17.0.2:47078]   File "/usr/lib/python2.7/site-packages/ipaserver/plugins/baseldap.py", line 1581, in execute
[Mon Oct 30 16:23:02.474833 2017] [wsgi:error] [pid 4859] [remote 172.17.0.2:47078]     delete_entry(pkey)
[Mon Oct 30 16:23:02.474835 2017] [wsgi:error] [pid 4859] [remote 172.17.0.2:47078]   File "/usr/lib/python2.7/site-packages/ipaserver/plugins/baseldap.py", line 1534, in delete_entry
[Mon Oct 30 16:23:02.474837 2017] [wsgi:error] [pid 4859] [remote 172.17.0.2:47078]     dn = callback(self, ldap, dn, *nkeys, **options)
[Mon Oct 30 16:23:02.474840 2017] [wsgi:error] [pid 4859] [remote 172.17.0.2:47078]   File "/usr/lib/python2.7/site-packages/ipaserver/plugins/host.py", line 842, in pre_callback
[Mon Oct 30 16:23:02.474842 2017] [wsgi:error] [pid 4859] [remote 172.17.0.2:47078]     certs = self.api.Command.cert_find(host=keys)['result']
[Mon Oct 30 16:23:02.474852 2017] [wsgi:error] [pid 4859] [remote 172.17.0.2:47078]   File "/usr/lib/python2.7/site-packages/ipalib/frontend.py", line 450, in __call__
[Mon Oct 30 16:23:02.474854 2017] [wsgi:error] [pid 4859] [remote 172.17.0.2:47078]     return self.__do_call(*args, **options)
[Mon Oct 30 16:23:02.474856 2017] [wsgi:error] [pid 4859] [remote 172.17.0.2:47078]   File "/usr/lib/python2.7/site-packages/ipalib/frontend.py", line 478, in __do_call
[Mon Oct 30 16:23:02.474859 2017] [wsgi:error] [pid 4859] [remote 172.17.0.2:47078]     ret = self.run(*args, **options)
[Mon Oct 30 16:23:02.474861 2017] [wsgi:error] [pid 4859] [remote 172.17.0.2:47078]   File "/usr/lib/python2.7/site-packages/ipalib/frontend.py", line 800, in run
[Mon Oct 30 16:23:02.474863 2017] [wsgi:error] [pid 4859] [remote 172.17.0.2:47078]     return self.execute(*args, **options)
[Mon Oct 30 16:23:02.474865 2017] [wsgi:error] [pid 4859] [remote 172.17.0.2:47078]   File "/usr/lib/python2.7/site-packages/ipaserver/plugins/cert.py", line 1588, in execute
[Mon Oct 30 16:23:02.474867 2017] [wsgi:error] [pid 4859] [remote 172.17.0.2:47078]     self.ca_enabled = self.api.Command.ca_is_enabled()['result']
[Mon Oct 30 16:23:02.474869 2017] [wsgi:error] [pid 4859] [remote 172.17.0.2:47078]   File "/usr/lib/python2.7/site-packages/ipalib/base.py", line 134, in __setattr__
[Mon Oct 30 16:23:02.474871 2017] [wsgi:error] [pid 4859] [remote 172.17.0.2:47078]     SET_ERROR % (self.__class__.__name__, name, value)
[Mon Oct 30 16:23:02.474873 2017] [wsgi:error] [pid 4859] [remote 172.17.0.2:47078] AttributeError: locked: cannot set cert_find.ca_enabled to True
```

Reply:
Switched to using context instead. It is concerning that python3 allowed this though.